defaults/main.yml

```diff
@@ -1,513 +1,50 @@
-## General
-
-#redis_version: 2.8.7
-
-## Service options
-
-# Owner
-#redis_user: redis
-#redis_group: redis
-
-# start on boot
-redis_service_enabled: yes
+#### installation variables ####
+redis_user: redis
+redis_group: redis
+
+redis_install_path: "/redis" #redis source will be here
+redis_conf_path: "/conf/redis" #redis configuration and running instance data will be here
+redis_mode: simple # custer or simple
+redis_version: 4.0.1
 # current state: started, stopped
 redis_service_state: started
+#redis_cluster_replicas: 1
+redis_install_new_version: False # set this to true to force an installation
+# start on boot
+redis_service_enabled: yes
 
-# Files & Paths
-#redis_install_path: "/opt/redis"
-redis_conf_path: "/etc/redis"
-#redis_data_path: "{{ redis_install_path }}/data"
-#redis_log_path: "/var/log/redis"
```

Comment:
This is a NTH :) I prefer logs in an own folder. For this purpose you should create the folder:
- name: REDIS | Create log directory
  file:
    path: "{{ redis_log_path }}"
    state: directory
    owner: "{{ redis_user }}"
    group: "{{ redis_group }}"